.gitignore

```diff
@@ -1 +1,2 @@
 /build/
+/result
```

Comment:
There is an `-o` option for `nix-build` that lets the user define which output link should be used. Nix uses the `result` link to keep the build artifacts from being garbage collected. I will put the `-o` option in the docs, but the name will be under the control of the user. If they use `nix-build` instead of `nix-build -o build`, the link will be created and might be committed by uncareful developers. The motivation for this was similar to the build directory, to ensure it's not committed by mistake but I don't have strong feelings about keeping it in here.

Reply:
My perspective: If someone runs a command mentioned in our documentation (except something which intends to modify files, like `clang-format`), `git status` should report no changes or untracked files.

Adding `/result` to .gitignore is fine if using `-o` is unnatural. But if `-o` is typical (like it is with CMake), I think we should use it in documentation.

Reply:
I do not like putting `nix-build -o build` into the documentation because it creates a link to a directory in /nix/store. This link overlaps with the default build directory we mention in the howtos, so not having write access to that directory might be super confusing.

The natural way to build a nix project is by using `nix-build` which will end up with a `result` links (as described). Looking at the whole picture together, I have decided to leave it in. But as said, I have no strong feelings about this so if you change your mind in the future, feel free to get rid of it.